### PR TITLE
Remove all suffixes in FSM diagram guards

### PR DESCRIPTION
**Pull request by raymond212**
### Description

Small fix for FSM diagram guards.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/software/ai/fsm_diagram_generator.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def remove_suffix(str, suffix):
         # Remove suffixes from states, guard, action
         src_state = remove_suffix(src_state, "_S")
         dest_state = remove_suffix(dest_state, "_S")
-        guard = remove_suffix(guard, "_G")
+        guard = "&&".join([remove_suffix(g, "_G") for g in guard.split("&&")])
```
Comment (williamckha) [inline src/software/ai/fsm_diagram_generator.py]: Can we try doing `" && ".join`

Comment (raymond212) [reply src/software/ai/fsm_diagram_generator.py]: Good idea, that'll look nicer visually. Done.

**Review by williamckha — APPROVED** [review]

**Comment by williamckha** [comment]
Pinging @nimazareian @itsarune for approval

**Review by itsarune — APPROVED** [review]
LGTM!

**Pull request merged**